### ISSUE TITLE
Bolt 2 style node

### ISSUE THREAD

**Issue by SadE54** [thedmd/imgui-node-editor#124]
Hi,

With a lot of nodes connections can become illegible.
Looking at different existing systems, it seems that bolt 2 found (at least partially) a solution : 
have the control flow inputs/outputs placed at the top and bottom of the node, and data horizontally
if there's several control inputs for exemple, the horizontal line is splitted accordingly.
I don't know if it's currently doable with the editor ?

![chrome_j6hRwlgLhD](https://user-images.githubusercontent.com/2878768/133579350-05bffe04-964a-4cfe-bdaa-0db8ddcddd9f.png)

**Comment by thedmd** [comment]
Yes, it is possible. Editor allow you to place pins anywhere on the node (or even outside of the node). Every pin can have different input/output direction, so by default output can point down, inputs can point up. Blueprint example have example of such connection in behavioral node style.
